src/ProjectService.Business/Commands/Image/CreateImageCommand.cs

```diff
@@ -12,123 +16,125 @@
+        .ToList();
+
+      string logMessage = $"Errors while creating images for user id {userId}.";
+
+      try
+      {
+        IOperationResult<ICreateImagesResponse> response = _rcImages.GetResponse<IOperationResult<ICreateImagesResponse>>(
+          ICreateImagesRequest.CreateObj(images, ImageSource.Project)).Result.Message;
 
-        public CreateImageCommand(
-            IImageRepository repository,
-            IRequestClient<ICreateImagesRequest> rcImages,
-            ILogger<CreateImageCommand> logger,
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IDbImageMapper dbProjectImageMapper,
-            ICreateImageValidator validator)
+        if (response.IsSuccess && response.Body.ImagesIds != null)
         {
-            _repository = repository;
-            _rcImages = rcImages;
-            _logger = logger;
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-            _dbProjectImageMapper = dbProjectImageMapper;
-            _validator = validator;
+          return response.Body.ImagesIds;
         }
 
-        public OperationResultResponse<List<Guid>> Execute(CreateImageRequest request)
-        {
-            OperationResultResponse<List<Guid>> response = new();
+        _logger.LogWarning(
+          logMessage + "Errors: { Errors}",
+          string.Join('\n', response.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage);
+      }
+
+      errors.Add("Can not create images. Please try again later.");
+
+      return null;
+    }
+
+    public CreateImageCommand(
+      IImageRepository repository,
+      IRequestClient<ICreateImagesRequest> rcImages,
+      ILogger<CreateImageCommand> logger,
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IDbImageMapper dbProjectImageMapper,
+      ICreateImageValidator validator,
+      IUserRepository userRepository)
+    {
+      _repository = repository;
+      _rcImages = rcImages;
+      _logger = logger;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _dbProjectImageMapper = dbProjectImageMapper;
+      _validator = validator;
+      _userRepository = userRepository;
+    }
 
-            if (!_accessValidator.HasRights(Rights.AddEditRemoveProjects))
-            {
-                _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+    public OperationResultResponse<List<Guid>> Execute(CreateImageRequest request)
+    {
+      OperationResultResponse<List<Guid>> response = new();
 
-                response.Status = OperationResultStatusType.Failed;
-                response.Errors.Add("Not enough rights.");
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
+      if (!_accessValidator.HasRights(Rights.AddEditRemoveProjects)
```

Comment:
if ImageType.Project,  projectAdmin can add/remove images

Reply:
this check is correct? if i have rights and imageType == Task but project already Exist - check is failed and i tried to create entity that already exist.

Reply:
done